query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -186,31 +177,26 @@ public QueryCost canHandle(QueryRequest request, Set<String> referencedSources,
    * @return True if the given filter node matches all the view filters from the given map, False
    * otherwise.
    */
-  private boolean doViewFiltersMatch(Filter filter, Map<String, ViewColumnFilter> viewFilterMap) {
-    // 1. Basic case: Filter is a leaf node. This will be a match only if there is only one
-    // view filter and that matches the leaf filter.
+  private Set<String> getMatchingViewFilterColumns(Filter filter, Map<String,
```

Comment:
beautiful!

Reply:
Thanks for the thorough review and ideas.